Gifski/Gifski.swift

```diff
@@ -4,53 +4,54 @@ import AVFoundation
 final class Gifski {
 	private(set) var isRunning = false
 	private var progress: Progress!
+    private var observation: NSKeyValueObservation?
 
 	// `progress.fractionCompleted` is KVO-compliant, but we expose this for convenience
-	var onProgress: ((_ progress: Double) -> Void)?
+	var onProgress: ((_ progress: Progress) -> Void)?
 
 	/**
 	- parameters:
 		- frameRate: Clamped to 5...30. Uses the frame rate of `inputUrl` if not specified.
 	*/
-	@discardableResult
 	func convertFile(
 		_ inputUrl: URL,
 		outputUrl: URL,
 		quality: Double = 1,
 		dimensions: CGSize? = nil,
 		frameRate: Int? = nil
-	) -> Progress {
+	) {
```

Comment:
Per https://github.com/sindresorhus/gifski-app/issues/7#issuecomment-368352717, you can drop the `isRunning` variable if you can think of a better way to check if the conversion is already running.

Reply:
@LarsJK Did you think of a way to remove `isRunning`? I guess we could check that `process != nil` and then check `!process.isFinished`.

Reply:
No I haven't looked into it yet, but that sounds like a good plan. Maybe make isRunning a computed property based on that?

Is there a reason why we can't run 2 conversions at the same time? If it's just a UI thing and not a restriction of this function maybe the isRunning check belongs in AppDelegate instead?

Reply:
> Maybe make isRunning a computed property based on that?

👍 

> Is there a reason why we can't run 2 conversions at the same time?

No, it's just that calling the method twice on the same instance will override the current instance variables. I'm open to ideas on how to improve this. I considered moving from `convertFile` to just making it an `init()`, so it's not possible to accidentally call it twice on the same instance, but then it would not be possible to add a method in the future to convert from/to `Data`, for example. Do you know any better way to handle this?

Reply:
What about making static factory methods that call, a private init and convert. Could that work?

Reply:
Yeah. That’s a great idea. Go for it. :)